scripts/lib/CIME/namelist.py

```diff
@@ -171,6 +171,20 @@ def is_valid_fortran_name(string):
     """
     return FORTRAN_NAME_REGEX.search(string) is not None
 
+def fortran_name_index(string):
+    """ Get name index if provided
+
+    >>> fortran_name_index("foo(2)")
+    2
+    >>> fortran_name_index("bar")
+
+    >>> fortran_name_index("foo(a)")
+
+    """
+    m = FORTRAN_NAME_REGEX.search(string)
+    if m is not None and m.group(2) is not None:
+        return int(m.group(2))
+
```

Comment:
Returning an int here is restrictive in that you will not be able to handle a list of ints, such as created by an array section. Of course, even a list will not always help (see below).